src/ctypes/ctypes_memory.ml

```diff
@@ -223,6 +223,14 @@ struct
     set arr len '\x00';
     arr
 
+  let to_string a =
+    let len = length a in
+    let bytes = Bytes.create len in
+    for i = 0 to len - 1 do
+        Bytes.set bytes i (unsafe_get a i)
+    done;
+    Bytes.unsafe_to_string bytes
```

Comment:
I think this implementation could be reduced to a single line with [`String.init`](https://github.com/ocaml/ocaml/blob/1fd6dc077d63fc8e0b36e2a684dae55a0f3383d4/stdlib/string.mli#L84-L92), something like (untested):
```ocaml
let to_string a = String.init (length a) (unsafe_get a)
```